content/300-guides/500-other/900-advanced-database-tasks/06-typesafe-raw-sql/01-safeql-gis.mdx

```diff
@@ -0,0 +1,234 @@
+
+### Add <inlinecode>@ts-safeql/eslint-plugin</inlinecode> to your ESLint plugins
+
+You should now add `@ts-safeql/eslint-plugin` to your list of ESLint plugins. In our example we are using an `.eslintrc.js` file, but this can be applied to [any way that you configure ESLint](https://eslint.org/docs/latest/use/configure/)
+
+```js file=.eslintrc.js highlight=3
+/** @type {import('eslint').Linter.Config} */
+module.exports = {
+  "plugins": [..., "@ts-safeql/eslint-plugin"],
+  ...
+}
+```
+
+### Add <inlinecode>@ts-safeql/check-sql</inlinecode> rules
+
+Now setup rules so that SafeQL will mark invalid SQL queries as ESLint errors
+
+```js file=.eslintrc.js highlight=4-22;add
+/** @type {import('eslint').Linter.Config} */
+module.exports = {
+  plugins: [..., '@ts-safeql/eslint-plugin'],
+  rules: {
+    '@ts-safeql/check-sql': [
+      'error',
+      {
+        connections: [
+          {
+            // The migrations path:
+            migrationsDir: './prisma/migrations',
+            targets: [
+              // This makes `prisma.$queryRaw` and `prisma.$executeRaw` commands linted
+              { tag: 'prisma.+($queryRaw|$executeRaw)', transform: '{type}[]' },
+              // This makes `client.$queryRaw` and `client.$executeRaw` commands linted
+              { tag: 'client.+($queryRaw|$executeRaw)', transform: '{type}[]' },
+            ],
+          },
+        ],
+      },
+    ],
+  },
+}
+```
+
+### Connect to your database
+
+Finally, set up a `connectionUrl` for SafeQL so that it can read the state of your database, including columns for each table.
+
+Our example relies on `dotenv` to get the same connection string that is used by Prisma. We recommend this in order to keep your database URL out of version control.
+
+```js file=.eslintrc.js highlight=1,6-9,16;add
+require('dotenv').config()
+
+/** @type {import('eslint').Linter.Config} */
+module.exports = {
+  plugins: ['@ts-safeql/eslint-plugin'],
+  // exclude `parserOptions` if you are not using TypeScript
+  parserOptions: {
+    project: './tsconfig.json',
+  },
+  rules: {
+    '@ts-safeql/check-sql': [
+      'error',
+      {
+        connections: [
+          {
+            connectionUrl: process.env.DATABASE_URL,
+            // The migrations path:
+            migrationsDir: './prisma/migrations',
+            targets: [
+              // what you would like SafeQL to lint. This makes `prisma.$queryRaw` and `prisma.$executeRaw`
+              // commands linted
+              { tag: 'prisma.+($queryRaw|$executeRaw)', transform: '{type}[]' },
+            ],
+          },
+        ],
+      },
+    ],
+  },
+}
+```
+
+## 4. Creating an extension for geographic data
+
+Now let's make a Prisma Client extension in order to query this model. We will be making an extension that finds the closest points of interest to a given longitude and latitude.
+
+```ts file=lib/prisma.ts
+<!-- prettier-ignore-start -->
```

Comment:
why are these prettier-snippets included?

Reply:
@jharrell I'm inclined to delete these prettier snippets. Are they needed? If so, what for?